terraform/modules/cluster/addons.tf

```diff
@@ -474,17 +452,15 @@ resource "random_string" "fluentbit_log_group" {
 }
 
 locals {
-  oidc_url = replace(data.aws_eks_cluster.cluster.identity[0].oidc[0].issuer, "https://", "")
```

Comment:
the EKS module outputs an attribute [`oidc_provider`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/df1b62548c1d8c7117f4ab45c5b494de64b34cb8/outputs.tf#L106) which is the issuer url with the `https://` stripped. It is named `oidc_provider` because this is the name assigned in the console, regardless of whether the provider was created by EKS or not

[`cluster_oidc_issuer_url`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/df1b62548c1d8c7117f4ab45c5b494de64b34cb8/outputs.tf#L30) is the form with the fully qualified URL including `https://`

tl;dr - OIDC provider is whats used for IRSA, not the issuer URL